### PR TITLE
Add support to override the canonical url of a page

### PR DESCRIPTION
**Pull request by nerdcel**
Suggests that the canonical field is either of type text or url

### CODE REVIEW

File: classes/JohannSchopplich/Helpers/PageMeta.php [hunk 1]
```diff
@@ -109,7 +110,7 @@ public function robots(): string
 
         $html[] = Html::tag('link', null, [
             'rel' => 'canonical',
-            'href' => $this->page->url(),
+            'href' => $canonical->isEmpty() ? $this->page->url() : $canonical->value(),
```
Comment (johannschopplich) [inline classes/JohannSchopplich/Helpers/PageMeta.php]:
```suggestion
            'href' => $canonical->or($this->page->url())->value(),
```

Comment (johannschopplich) [reply classes/JohannSchopplich/Helpers/PageMeta.php]: Could be simplified.

**Review by johannschopplich — CHANGES_REQUESTED** [review]
Nice one! One idea tho. :)

File: classes/JohannSchopplich/Helpers/PageMeta.php (hunk 1 above)
Comment (nerdcel) [reply classes/JohannSchopplich/Helpers/PageMeta.php]: Good idea. I didn't think about that :)

**Comment by johannschopplich** [comment]
Thanks for the contribution!

**Pull request merged**